src/ucp/core/ucp_worker.c

```diff
@@ -328,10 +328,26 @@ static void ucp_worker_wakeup_cleanup(ucp_worker_h worker)
     }
 }
 
+static UCS_F_ALWAYS_INLINE
+int ucp_worker_iface_has_event_notify(const ucp_worker_iface_t *wiface)
+{
+    return (wiface->attr.cap.event_flags & UCT_IFACE_FLAG_EVENT_FD) ||
+           (wiface->attr.cap.event_flags & UCT_IFACE_FLAG_EVENT_ASYNC_CB);
+}
+
+static UCS_F_ALWAYS_INLINE
+int ucp_worker_iface_use_event_fd(const ucp_worker_iface_t *wiface)
+{
+    /* use iface's fd if it is supported by UCT iface and asynchronous
+     * callback mechanism isn't supported (this is preferred mechanism) */
```

Comment:
document that we prefer the async callback since it will be called anyway

Reply:
done